sentry_sdk/client.py

```diff
@@ -616,9 +623,18 @@ def capture_event(
             for attachment in attachments or ():
                 envelope.add_item(attachment.to_envelope_item())
 
+            if self.spotlight:
+                self.spotlight.capture_envelope(envelope)
+
+            if self.transport is None:
+                return None
```

Comment:
This moved here otherwise we would've returned too early and the envelope wouldn't even be assembled. Yes it's less efficient but anyone using the SDK shouldn't notice anything.